src/main/java/org/assertj/assertions/generator/BaseAssertionGenerator.java

```diff
@@ -319,14 +322,49 @@ private String generateAssertionsEntryPointClassContent(final Set<ClassDescripti
         ? determineBestEntryPointsAssertionsClassPackage(classDescriptionSet)
         : entryPointClassPackage;
     entryPointAssertionsClassContent = replace(entryPointAssertionsClassContent, PACKAGE, classPackage);
-
+    Set<String> imports;
+    if (entryPointAssertionsClassContent.contains(IMPORTS)) {
+      imports = extractImports(classDescriptionSet);
+      entryPointAssertionsClassContent = replace(entryPointAssertionsClassContent, IMPORTS, listNeededImports(imports));
+    } else {
+      imports = new HashSet<>();
+    }
     String allEntryPointsAssertionContent = generateAssertionEntryPointMethodsFor(classDescriptionSet,
-                                                                                  entryPointAssertionMethodTemplate);
+                                                                                  entryPointAssertionMethodTemplate,
+                                                                                  imports);
     entryPointAssertionsClassContent = replace(entryPointAssertionsClassContent, ALL_ASSERTIONS_ENTRY_POINTS,
                                                allEntryPointsAssertionContent);
     return entryPointAssertionsClassContent;
   }
 
+  private static String listNeededImports(Collection<String> imports) {
```

Comment:
since it has been decided to go with the imports, I'm ok with the breaking change, users that have used their own templates would only have to add `${imports}` at the top, seems reasonable to me if we document that clearly.